### ISSUE TITLE
Not aware of rawtherapee and darktable when all is installed as flatpak

### ISSUE THREAD

**Issue by Yajo** [flathub/org.gimp.GIMP#138]
I tried to open a raw picture.

A dialog appeared saying:

> Opening '/var/home/yajo/Imágenes/Fotos compartidas/2022-02-19 Sesión de fotos/IMG_9528.CR2' failed: There is no RAW loader installed to open 'Raw Canon' files.
>
> GIMP currently supports these RAW loaders:
> - darktable (http://www.darktable.org/), at least 1.7
> - RawTherapee (http://rawtherapee.com/), at least 5.2
> 
> Please install one of them in order to load RAW files.

I have both of those apps installed, but they are flatpaks, just like GIMP.

So GIMP flatpak should be somehow aware of these other apps when all is installed as flatpaks AFAICS.

**Comment by Jehan** [comment]
Yes by sandbox-logic, a software doesn't see outside its box, and in particular, it doesn't see other software. For such features, it requires "portals", but there is currently no portal to run other flatpaks.

I know there is a report somewhere about making such a portal. I can't find the link right now. Anyway the day when it will become a reality, we'll have to use this to allow this feature. In the meantime, it is just not possible by technology limitation.

**Comment by Yajo** [comment]
However there must be a way, don't you think? because Flatseal can see what other flatpak apps exist installed, and even manage their permissions.

I guess It'll be around this: https://github.com/flathub/com.github.tchx84.Flatseal/blob/9156b312e6fa1a9087e8a3746fe6c48715ae1b8a/com.github.tchx84.Flatseal.json#L13-L14

**Comment by Jehan** [comment]
@Yajo As you can see from the link you give, the way flatseal works (which was also what I understood reading info about it, though I never looked in the code) is that they have access to the flatpak directories directly. This is even very precisely the 2 lines you give yourself.

This has absolutely no interest for us, unless we also wanted to edit the other flatpak-s directly. It doesn't give us magically the ability to run them, then get returned files back from these software. Or well… if it did, there would be quite a serious leak to fix!

Flatseal does not run other flatpak, it doesn't interact with them. It only modifies their files.

**Comment by Yajo** [comment]
Ah I see... Then maybe GIMP should use the open portal directly or something like that?

For example, I installed [Evolution](https://flathub.org/apps/details/org.gnome.Evolution). When an email has an attached image, I can double-click on it, and a dialog will appear asking me with which app I want to open it. It lists all apps that open images, no matter if flatpak or not:

![imagen](https://user-images.githubusercontent.com/973709/155083602-dcee2539-51d2-4e86-9fc5-e40119124eaa.png)

But then I guess that should be upstream issue, right?

**Comment by Jehan** [comment]
This is again a different concept. It allows to open an image (or any file) in another software, and then it's over. What we do in GIMP is a roundtrip as part of a workflow. We send file to raw image developers (darktable, RawTherapee) then once the work there is done, it comes back in GIMP to be able to continue the work there. It's a collaboration mechanism to make use of the power of several powerful application in a more flowing process.

Just opening it in another software (then the image is gone) while it is opened in GIMP doesn't make really much sense. Just open it directly in darktable or RawTherapee if that's what you want. ;-)

**Comment by Yajo** [comment]
Ok that's what I did. I thought there was some deeper integration I was missing 😁
Thanks!

**Comment by Jehan** [comment]
Let me reopen this report as a reminder though. We still want the collaboration process to be possible, it's simply not possible right now as far as we know. So it's good we have a tracking report about this issue.

**Comment by heyakyra** [comment]
I believe this actually _is_ possible, however it would require re-packaging the raw-loaders as add-ons, similar to the many other addons that already exist:

```
$ flatpak install gimp
Looking for matches…
Remotes found with refs similar to ‘gimp’:

   1) ‘fedora’ (system)
   2) ‘flathub’ (system)
   3) ‘flathub-beta’ (system)
   4) ‘flathub-beta’ (user)

Which do you want to use (0 to abort)? [0-4]: 2
Similar refs found for ‘gimp’ in remote ‘flathub’ (system):

   1) runtime/org.gimp.GIMP.Plugin.Fourier/x86_64/2-40
   2) runtime/org.gimp.GIMP.Plugin.Fourier/x86_64/2-3.36
   3) runtime/org.gimp.GIMP.HEIC/x86_64/stable
   4) runtime/org.gimp.GIMP.Plugin.FocusBlur/x86_64/2-40
   5) runtime/org.gimp.GIMP.Plugin.GMic/x86_64/2-40
   6) runtime/org.gimp.GIMP.Plugin.BIMP/x86_64/2-3.36
   7) runtime/org.gimp.GIMP.Plugin.Resynthesizer/x86_64/2-3.36
   8) runtime/org.gimp.GIMP.Plugin.GMic/x86_64/3
   9) runtime/org.gimp.GIMP.Plugin.Resynthesizer/x86_64/2-40
  10) runtime/org.gimp.GIMP.Plugin.Fourier/x86_64/3
  11) runtime/org.gimp.GIMP.Plugin.FocusBlur/x86_64/2-3.36
  12) runtime/org.gimp.GIMP.Plugin.Resynthesizer/x86_64/3
  13) runtime/org.gimp.GIMP.Plugin.BIMP/x86_64/2-40
  14) runtime/org.gimp.GIMP.Plugin.LiquidRescale/x86_64/2-3.36
  15) runtime/org.gimp.GIMP.Plugin.LiquidRescale/x86_64/2-40
  16) app/org.gimp.GIMP/x86_64/stable
  17) runtime/org.gimp.GIMP.Plugin.GMic/x86_64/2-3.36
  18) runtime/org.gimp.GIMP.Manual/x86_64/2.10
  19) runtime/org.gimp.GIMP.Plugin.Lensfun/x86_64/2-3.36
  20) runtime/org.gimp.GIMP.Plugin.Lensfun/x86_64/2-40

Which do you want to use (0 to abort)? [0-20]:
```
